src/components/Requirements/CompletedSubReqCourse.vue

```diff
@@ -73,22 +68,16 @@ export default Vue.extend({
         return { semesterLabel: 'Transfer Credits', color: transferCreditColor, uniqueID: 0 };
       }
 
-      const courseTakenCode = `${this.courseTaken.subject} ${this.courseTaken.number}`;
+      const course = store.state.derivedCoursesData.courseMap[this.courseTaken.uniqueId];
+      const courseSemester =
+        store.state.derivedCoursesData.courseToSemesterMap[this.courseTaken.uniqueId];
+      const courseColor = course != null ? course.color : '';
+      const semesterLabel =
+        courseSemester != null
+          ? `${courseSemester.type} ${courseSemester.year}`
+          : `${getCurrentSeason()} ${getCurrentYear()}`;
 
-      for (let i = 0; i < this.semesters.length; i += 1) {
-        const semester = this.semesters[i];
-        const filteredSemesterCourses = semester.courses.filter(
-          course => course.crseId === this.courseTaken.courseId && course.code === courseTakenCode
-        );
-        if (filteredSemesterCourses.length > 0) {
-          return {
-            semesterLabel: `${semester.type} ${semester.year}`,
-            uniqueID: filteredSemesterCourses[0].uniqueID,
-            color: filteredSemesterCourses[0].color,
-          };
-        }
-      }
-      return { semesterLabel: `${getCurrentSeason()} ${getCurrentYear()}`, uniqueID: 0, color: '' };
+      return { semesterLabel, uniqueID: this.courseTaken.uniqueId, color: courseColor };
```

Comment:
Is there a need to return `uniqueID` here? I don't see `courseInfoAndSemesterLabel.uniqueID` being accessed.